NYPLAudiobookToolkit/DRM/FeedbookDRMProcessor.swift

```diff
@@ -0,0 +1,139 @@
+        
+        return true
+    }
+    
+    class func performAsyncDrm(book: OpenAccessAudiobook, drmData: [String: Any]) {
+        if let licenseCheckUrl = drmData["licenseCheckUrl"] as? URL {
+            weak var weakBook = book
+            URLSession.shared.dataTask(with: licenseCheckUrl) { (data, response, error) in
+                // Errors automatically mean success
+                if error != nil {
+                    weakBook?.drmStatus = .succeeded
+                    ATLog(.debug, "feedbooks::performAsyncDrm licenseCheck skip due to error: \(error!)")
+                    return
+                }
+                
+                // Explicitly check status value
+                if data != nil {
+                    if let jsonObj = try? JSONSerialization.jsonObject(with: data!, options: JSONSerialization.ReadingOptions()) as? [String: Any],
+                        let statusString = jsonObj?["status"] as? String {
+                        if statusString != "ready" && statusString != "active" {
+                            ATLog(.debug, "feedbooks::performAsyncDrm licenseCheck failed: \((try? JSONUtils.canonicalize(jsonObj: jsonObj) as String) ?? "")")
+                            weakBook?.drmStatus = .failed
+                            return
+                        }
+                    }
+                }
+
+                // Fallthrough on all other cases
+                weakBook?.drmStatus = .succeeded
+                ATLog(.debug, "feedbooks::performAsyncDrm licenseCheck fallthrough")
+            }
+        } else {
+            book.drmStatus = .succeeded
+            ATLog(.debug, "feedbooks::performAsyncDrm licenseCheck not needed")
+        }
+    }
+    
+    class func getFeedbookSecret(profile: String) -> String {
+        let tag = "feedbook_drm_profile_\(profile)"
+        let tagData = tag.data(using: .utf8)!
+        let query: [String: Any] = [
+            kSecClass as String: kSecClassKey,
+            kSecAttrApplicationTag as String: tagData,
+            kSecReturnData as String: true
+        ]
+        var item: AnyObject?
+        let status = SecItemCopyMatching(query as CFDictionary, &item)
+        if status == errSecSuccess {
+            if item == nil {
+                ATLog(.error, "Keychain item is nil for profile: \(profile)")
+            } else if let sItem = item as? String {
+                return sItem
+            } else if let dItem = item as? Data {
+                ATLog(.warn, "Keychain item was data instead of string for profile: \(profile)")
+                return String.init(data: dItem, encoding: .utf8) ?? ""
+            } else {
+                ATLog(.error, "Keychain item unknown error for profile: \(profile)")
+            }
+        } else {
+            ATLog(.error, "Could not fetch keychain item for profile: \(profile)")
+        }
+        return ""
+    }
+    
+    class func getJWTToken(profile: String, resourceUri: String) -> String {
```

Comment:
There's no reason. I thought it might be preferred to use an explicitly present value than complicate code with an optional. If optionals are preferred I can change this.